src/StructuredLogger/ObjectModel/TreeNode.cs

```diff
@@ -47,6 +47,8 @@ public bool IsExpanded
             }
         }
 
+        private Dictionary<ChildrenCacheKey, object> childrenCache;

```

Comment:
I can add and abstraction and two implementations - one array based and one list based.